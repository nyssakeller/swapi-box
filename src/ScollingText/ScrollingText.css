body {
  /* Force the body to fill the entire screen */
  width: 100%;
  height: 100%;
  /* Hide elements that flow outside the viewable space */
  
  /* Black background for the screen */
}
.scrolling-text {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.crawl {
  /* Position the element so we can adjust the top property in the animation */
  position: relative;
  /* Making sure the text is fully off the screen at the start and end of the animation */
  top: -100px;
  /* Defines the skew origin at the very center when we apply transforms on the animation */
  transform-origin: 50% 100%;

  /* Defines the skew origin at the very center when we apply transforms on the animation */
  /* Adds the crawl animation, which plays for one minute */
  animation: crawl 60s linear;
}

/*.scrolling-text p {
  font-size: 60px;
}
*/

@keyframes crawl {
  0% {
    /* The element starts below the screen */
    top: 0;
    /* Rotate the text 20 degrees but keep it close to the viewer */
    transform: rotateX(20deg) translateZ(0);
  }
  100% { 
    /* This is a magic number, but using a big one to make sure the text is fully off the screen at the end */
    top: -6000px;
    /* Slightly increasing the rotation at the end and moving the text far away from the viewer */
    transform: rotateX(25deg) translateZ(-2500px);
  }
}
