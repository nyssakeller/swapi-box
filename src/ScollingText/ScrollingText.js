/* eslint-disable */
import React from 'react';
import './ScrollingText.css';
import PropTypes from 'prop-types';

const ScrollingText = ({filmData}) => {
  const index = Math.floor((Math.random() * 6) + 1);

  return (
    <section className='scrolling-text'>
      <h1>{filmData[index].title}</h1>
      <p>{filmData[index].description}</p>
    </section>
  );
};

export default ScrollingText;

ScrollingText.propTypes = {
  filmData: PropTypes.object.isRequired
};