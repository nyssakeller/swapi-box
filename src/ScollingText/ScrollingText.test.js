/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import ScrollingText from './ScrollingText.js';

const mockFilm = [
  {
    title: 'a new hope',
    opening_crawl: 'text',
    release_date: 'yesterday'
  },
  {
    title: 'another title',
    opening_crawl: 'more text',
    release_date: 'tomorrow'
  },
  {
    title: 'another title',
    opening_crawl: 'more text',
    release_date: 'tomorrow'
  },
  {
    title: 'another title',
    opening_crawl: 'more text',
    release_date: 'tomorrow'
  },
  {
    title: 'another title',
    opening_crawl: 'more text',
    release_date: 'tomorrow'
  },
  {
    title: 'another title',
    opening_crawl: 'more text',
    release_date: 'tomorrow'
  },
  {
    title: 'another title',
    opening_crawl: 'more text',
    release_date: 'tomorrow'
  },
];


describe('ScrollingText', () => {
 
  it('snapshot test', () => {
    const wrapper = shallow(<ScrollingText filmData={mockFilm} />)
    expect(wrapper).toMatchSnapshot();
  });

});