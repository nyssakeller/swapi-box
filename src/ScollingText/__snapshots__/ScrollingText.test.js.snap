// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollingText
    filmData={
        Array [
            Object {
              "opening_crawl": "text",
              "release_date": "yesterday",
              "title": "a new hope",
            },
            Object {
              "opening_crawl": "more text",
              "release_date": "tomorrow",
              "title": "another title",
            },
            Object {
              "opening_crawl": "more text",
              "release_date": "tomorrow",
              "title": "another title",
            },
            Object {
              "opening_crawl": "more text",
              "release_date": "tomorrow",
              "title": "another title",
            },
            Object {
              "opening_crawl": "more text",
              "release_date": "tomorrow",
              "title": "another title",
            },
            Object {
              "opening_crawl": "more text",
              "release_date": "tomorrow",
              "title": "another title",
            },
            Object {
              "opening_crawl": "more text",
              "release_date": "tomorrow",
              "title": "another title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          another title
</h1>,
        <p />,
      ],
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "another title",
        },
        "ref": null,
        "rendered": "another title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            another title
</h1>,
          <p />,
        ],
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "another title",
          },
          "ref": null,
          "rendered": "another title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
